src/coreclr/jit/codegenxarch.cpp

```diff
@@ -3178,14 +3163,24 @@ void CodeGen::genCodeForInitBlkUnroll(GenTreeBlk* node)
             regSize = min(regSize, YMM_REGSIZE_BYTES);
         }
 
-        bool zeroing = false;
         if (src->gtSkipReloadOrCopy()->IsIntegralConst(0))
         {
             // If the source is constant 0 then always use xorps, it's faster
             // than copying the constant from a GPR to a XMM register.
             emit->emitIns_SIMD_R_R_R(INS_xorps, EA_ATTR(regSize), srcXmmReg, srcXmmReg, srcXmmReg);
-            zeroing = true;
         }
+#ifdef FEATURE_SIMD
+        else if (src->gtSkipReloadOrCopy()->IsIntegralConst())
+        {
+            // Populate a constant vector from the fill value and save it
+            // to the data section so we can load it by address.
+            assert(regSize <= ZMM_REGSIZE_BYTES);
+            simd64_t constValue;
+            memset(&constValue, (uint8_t)(src->AsIntCon()->IconValue() & 0xFF), sizeof(simd64_t));
```

Comment:
Is it worth asserting that the underlying constant is actually the same in all bytes?

Reply:
Probably? But we already do the same in lower - https://github.com/dotnet/runtime/blob/main/src/coreclr/jit/lowerxarch.cpp#L331-L338C61